woocommerce/woocommerce-ios
# Stripe Terminal SDK update 3.3.1

## Description

- Updated Stripe Terminal SDK to [version](https://github.com/stripe/stripe-terminal-ios/releases) 3.3.1 
- Discussion: pfoUAQ-hK-p2
- Plan is to also use "Added network and wallet to SCPCardPresentDetails." change to add these parameters to tracking events

## Changes in SDK from current version (3.1.0)

- Versions 3.3.1, 3.2.1 just have fixes in them

### 3.2.0
* Update: retrievePaymentIntent and retrieveSetupIntent no longer require a connected reader.
* Update: Adds [SCPCharge](https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPCharge.html).authorizationCode to the sdk's SCPCharge model when it is available.
    * Note for internet reader integrations, this feature requires [reader software version](https://stripe.com/docs/terminal/readers/bbpos-wisepos-e#reader-software-version) 2.18 or later to be installed on your internet reader.
* Update: Added network and wallet to SCPCardPresentDetails.
* Update: Added paymentMethodId to SCPPaymentIntent.
* Update: The amount of time a reader can be used offline before needing to be activated online has been reduced to 30 days.
* Update: SCPReaderNetworkStatus has a new case (SCPReaderStatusUnknown) that represents the network status of an internet reader that was discovered while the SDK is operating in offline mode.

### 3.3.0
* New: [rebootReader](https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPTerminal.html#/c:objc(cs)SCPTerminal(im)rebootReader:) method to reboot Bluetooth readers.
* New: An optional [reader:didDisconnect:](https://stripe.dev/stripe-terminal-ios/docs/Protocols/SCPBluetoothReaderDelegate.html#/c:objc(pl)SCPBluetoothReaderDelegate(im)reader:didDisconnect:) method was added to [SCPBluetoothReaderDelegate](https://stripe.dev/stripe-terminal-ios/docs/Protocols/SCPBluetoothReaderDelegate.html#/c:objc(pl)SCPBluetoothReaderDelegate) which provides a new [SCPDisconnectReason](https://stripe.dev/stripe-terminal-ios/docs/Enums/SCPDisconnectReason.html) to communicate known reasons why a Bluetooth reader disconnected.
* New: Support refunding payments with the SCPPaymentIntent.stripeId.
    * Note for internet reader integrations, this feature requires [reader software version](https://stripe.com/docs/terminal/readers/bbpos-wisepos-e#reader-software-version) 2.19 or later to be installed on your internet reader.
* New: Added support for retrieving and updating reader settings on WisePOS E and Stripe S700 by calling retrieveReaderSettings and setReaderSettings on SCPTerminal.
    * Beta: Accessibility settings are provided at this time, allowing text-to-speech via speakers to be turned on and off as needed.
* Beta: Added a [collectInputs](https://stripe.com/docs/terminal/features/collect-inputs) method to display forms and collect information from customers.

## Testing instructions


Used https://github.com/woocommerce/woocommerce-ios/pull/11080 for reference

Test any combination of card reader connection, and payment flows that worked previously using card present or tap to pay. The same behaviour should be expected after the update. As a guide, on my testing I went through the following scenarios:

* Connection from Order payment flow
* Connection from Payments menu
* Connection of a reader with a mandatory update
* Connection of M2 reader
* Connection from switched off state

I was not able to test the following, since I don't have the option:
* Connection of Chipper reader
* Take Interac payment
* Refund Interac payment
* Connection of WisePad 3 reader
* Take payment after in-line connection process
* Take payment with a previously-connected reader
* Take contactless payment
* Take card-inserted payment
* Reader update: error message when attempted with low battery
* Reader update: successfully updates software and starts payment (in that flow) when completed

---
